dockerfiles/rust-builder/Dockerfile

```diff
@@ -34,14 +32,17 @@ ADD "https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-in
 
 # install tools and dependencies
 RUN set -eux; \
-	echo "deb http://ppa.launchpad.net/ubuntu-mozilla-security/ppa/ubuntu bionic main" \
-		> /etc/apt/sources.list.d/mozilla.list; \
 	apt-get -y update; \
 	apt-get install -y --no-install-recommends \
-		libssl-dev clang lld libclang-dev cmake make firefox \
+		libssl-dev clang lld libclang-dev cmake make firefox-esr \
 		git pkg-config curl time rhash ca-certificates; \
+# install geckodriver
+	curl -s https://api.github.com/repos/mozilla/geckodriver/releases/latest \
+		| grep 'browser_download_url.*linux64.tar.gz"' \
+		| cut -d : -f 2,3 | tr -d \" | xargs -n 1 curl -sSL \
+		| tar -xzC /usr/bin; \
 # set a link to clang
-	ln -sf /usr/bin/clang /usr/bin/cc;\
+	ln -sf /usr/bin/clang /usr/bin/cc; \
```

Comment:
`update-alternatives --install /usr/bin/cc cc /usr/bin/clang 100` should be prefered